azurecompute/src/main/java/org/jclouds/azurecompute/compute/functions/DeploymentToNodeMetadata.java

```diff
@@ -16,16 +16,99 @@
  */
 package org.jclouds.azurecompute.compute.functions;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+import java.util.Map;
+import java.util.Set;
+
+import javax.inject.Inject;
+
 import org.jclouds.azurecompute.domain.Deployment;
+import org.jclouds.collect.Memoized;
 import org.jclouds.compute.domain.NodeMetadata;
+import org.jclouds.compute.domain.NodeMetadataBuilder;
+import org.jclouds.compute.functions.GroupNamingConvention;
+import org.jclouds.domain.Credentials;
+import org.jclouds.domain.Location;
 
 import com.google.common.base.Function;
+import com.google.common.base.Supplier;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
 
 public class DeploymentToNodeMetadata implements Function<Deployment, NodeMetadata> {
 
+	private final Supplier<Set<? extends Location>> locations;
+	private final GroupNamingConvention nodeNamingConvention;
+	private final OSImageToImage osImageToImage;
+	private final RoleSizeToHardware roleSizeToHardware;
+	private final Map<String, Credentials> credentialStore;
+
+	public static final Map<Deployment.Status, NodeMetadata.Status> serverStateToNodeStatus = ImmutableMap
+			  .<Deployment.Status, NodeMetadata.Status> builder()
+			  .put(Deployment.Status.DELETING, NodeMetadata.Status.PENDING)
+			  .put(Deployment.Status.SUSPENDED_TRANSITIONING, NodeMetadata.Status.PENDING)
+			  .put(Deployment.Status.RUNNING_TRANSITIONING, NodeMetadata.Status.PENDING)
+			  .put(Deployment.Status.DEPLOYING, NodeMetadata.Status.PENDING)
+			  .put(Deployment.Status.STARTING, NodeMetadata.Status.PENDING)
+			  .put(Deployment.Status.SUSPENDED, NodeMetadata.Status.SUSPENDED)
+			  .put(Deployment.Status.RUNNING, NodeMetadata.Status.RUNNING)
+			  .put(Deployment.Status.UNRECOGNIZED, NodeMetadata.Status.UNRECOGNIZED).build();
+
+	@Inject
+	DeploymentToNodeMetadata(@Memoized Supplier<Set<? extends Location>> locations,
+										GroupNamingConvention.Factory namingConvention, OSImageToImage osImageToImage,
+										RoleSizeToHardware roleSizeToHardware, Map<String, Credentials> credentialStore) {
+		this.nodeNamingConvention = checkNotNull(namingConvention, "namingConvention").createWithoutPrefix();
+		this.locations = checkNotNull(locations, "locations");
+		this.osImageToImage = checkNotNull(osImageToImage, "osImageToImage");
+		this.roleSizeToHardware = checkNotNull(roleSizeToHardware, "roleSizeToHardware");
+		this.credentialStore = checkNotNull(credentialStore, "credentialStore cannot be null");
```

Comment:
Since the constructor is package private and only the Guice injector will be able to call it, you can remove all the null checks. The injector will already fail if any parameter is missing.